src/s11/handlers/modify_bearer_handler.c

```diff
@@ -63,6 +63,37 @@ modify_bearer_processing(struct MB_Q_msg *mb_msg)
 
 	ModifyBearerRequestMsgData msgData;
 	memset(&msgData, 0, sizeof(msgData));
+	struct TAI *tai = &(mb_msg->tai);
+	struct CGI *cgi = &(mb_msg->utran_cgi);
+
+	msgData.servingNetworkIePresent = mb_msg->servingNetworkIePresent;
+	msgData.servingNetwork.mccDigit1 = tai->plmn_id.idx[0] & 0x0F;
+	msgData.servingNetwork.mccDigit2 = (tai->plmn_id.idx[0] & 0xF0) >> 4;
+	msgData.servingNetwork.mccDigit3 = tai->plmn_id.idx[1] & 0x0F;
+	msgData.servingNetwork.mncDigit1 = tai->plmn_id.idx[2] & 0x0F;
+	msgData.servingNetwork.mncDigit2 = (tai->plmn_id.idx[2] & 0xF0) >> 4;
+	msgData.servingNetwork.mncDigit3 = (tai->plmn_id.idx[1] & 0xF0) >> 4;
+
+	msgData.userLocationInformationIePresent = mb_msg->userLocationInformationIePresent;
```

Comment:
we are not using userLocationInformationIePresent field... same way servingNetworkIePresent.
If servingNetworkIePresent & userLocationInformationIePresent is present then only we should fill complete IE content.

Reply:
This is done by the gtp stack. However it will be good to have the check here itself. Will change this.